src/main/java/uk/gov/dwp/dataworks/errors/CurrentKeyIdFailure.java

```diff
@@ -0,0 +1,11 @@
+package uk.gov.dwp.dataworks.errors;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.SERVICE_UNAVAILABLE)
```

Comment:
Tied to the comment below about generic vs specific, but is the user passing a missing key a service unavailable or is it something like BAD_REQUEST because the user has passed in a piece of data that isn't valid.